packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1653,8 +1634,30 @@ You can use them by running Hardhat Network with 'hardfork' ${ACCESS_LIST_MIN_HA
 
     if (!this._common.gteHardfork(EIP155_MIN_HARDFORK)) {
       throw new InvalidArgumentsError(`Trying to send an EIP-155 transaction, but they are not supported by the current hardfork.  
-      
+
 You can use them by running Hardhat Network with 'hardfork' ${EIP155_MIN_HARDFORK} or later.`);
     }
   }
+
+  private _validateRawTransactionHardforkRequirements(rawTx: Buffer) {
+    if (rawTx[0] <= 0x7f && rawTx[0] !== 1 && rawTx[0] !== 2) {
+      throw new InvalidArgumentsError(`Invalid transaction`);
```

Comment:
I know this was just moved around, but can we use a more explicit error message here? Otherwise if a user runs into this, the cause is going to be harder to pin down.

Reply:
Done